src/App.js

```diff
@@ -0,0 +1,47 @@
+import logo from './logo.svg';
+import './App.css';
+import './TaskItem';
+import './DataContainer';
+import {useEffect, useState} from "react";
+import AppHeader from "./AppHeader";
+import TaskList from "./TaskList";
+
+function App(props) {
+    const [hideUncompleted, setHideUncompleted] = useState(true);
+    const [displayData, setDisplayData] = useState(props.data);
```

Comment:
All you need is:
displayData = hideCompleted ? props.data.filter(taskItem => !taskItem.isCompleted) : props.data

Rule is: if you can compute if from state or props, it's not state.